src/parser/expression.js

```diff
@@ -454,12 +457,13 @@ pp.parseExprAtom = function (refShorthandDefaultPos) {
         return this.finishNode(node, "DoExpression");
       }
 
-    case tt.regexp:
+    case tt.regexp: {
       const value = this.state.value;
-      node = this.parseLiteral(value.value, "RegExpLiteral");
+      const node = this.parseLiteral(value.value, "RegExpLiteral");
```

Comment:
A similar change could be made in most of the switch cases, but probably best to not make the diff larger